### PR TITLE
8288415: java/awt/PopupMenu/PopupMenuLocation.java is unstable in MacOS machines

### PR DESCRIPTION
**Pull request by manukumarvs**
java/awt/PopupMenu/PopupMenuLocation.java seems to be unstable in MacOS machines, especially in MacOSX 12 & 13 machines. It seems to be a testbug as adding some stability improvements fixes the issue. It intermittently fails in CI causing some noise. This test was already problem listed in windows due to an umbrella bug JDK-8238720. This fix doesn't cover the Windows issue, so the problem listing in windows will remain same.

Fix:
Some stability improvements have been done and the test has been run 100 times per platform in mach5 and got full PASS.

---------
### Progress
- [ ] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8288415](https://bugs.openjdk.org/browse/JDK-8288415): java/awt/PopupMenu/PopupMenuLocation.java is unstable in MacOS machines

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10606/head:pull/10606` \
`$ git checkout pull/10606`

Update a local copy of the PR: \
`$ git checkout pull/10606` \
`$ git pull https://git.openjdk.org/jdk pull/10606/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10606`

View PR using the GUI difftool: \
`$ git pr show -t 10606`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10606.diff">https://git.openjdk.org/jdk/pull/10606.diff</a>

</details>

### CODE REVIEW

File: test/jdk/java/awt/PopupMenu/PopupMenuLocation.java [hunk 1]
```diff
@@ -106,22 +119,37 @@ private void show(MouseEvent e) {
...
+                    "Failed, Not received the PopupMenu ActionEvent yet");
         }
         action = false;
     }
+
+    private static void captureScreen() {
+        Dimension screenSize = Toolkit.getDefaultToolkit().getScreenSize();
+        try {
+            ImageIO.write(robot.createScreenCapture(
+                    new Rectangle(0, 0, screenSize.width, screenSize.height)),
```
Comment (mrserb) [inline test/jdk/java/awt/PopupMenu/PopupMenuLocation.java:148]: This may capture the default screen while the frame is currently visible on another one.

Comment (manukumarvs) [reply test/jdk/java/awt/PopupMenu/PopupMenuLocation.java:148]: But, is there any way we can capture the current screen where the frame is displayed?

Comment (mrserb) [reply test/jdk/java/awt/PopupMenu/PopupMenuLocation.java:148]: You can pass coordinates/bounds of that screen to the createScreenCapture.

**Comment by aivanov-jdk** [comment]
You did something wrong, now there are 40 commits and 126 files changed. It doesn't look right.

Merging master should result in one commit only even though it brings multiple commits into the history.

**Comment by manukumarvs** [comment]
> You did something wrong, now there are 40 commits and 126 files changed. It doesn't look right.
> 
> Merging master should result in one commit only even though it brings multiple commits into the history.

Yea, I have seen that. I tried to fix it, but its not working, will delete this branch and create a new one.

**Pull request closed without merging**